NameMangling.md

```diff
@@ -0,0 +1,37 @@
+# Name mangling
+
+Linker will mangle names internally,
+but we're hoping to keep the mangling details
+out of the public api.
+
+See also renaming host visible names in [Visiblity](./Visiblity.md).
+
+(TBD)
+
+## Mangling
+To join multiple modules into one, we need to make sure that names do not collide. This is done by name mangling.
+
+Compiled WGSL files have a few exposed parts, namely the names of the entry functions and pipeline overridable constants. These need to be stable and predictable, so we will introduce a stable, predictable and human-readable name mangling scheme.
+
+Finally, only importable items need to be mangled.
+
+For name mangling, we introduce the concept of a **absolute module identifier**. This is an ID that is relative to the nearest _package_, and is unique for each module. For example, given a file `my/geom/sphere.wgsl`, and a package `my`, then the absolute module identifier is `my_geom_sphere`. It is always this, regardless of how the file is imported. If a file or folder name already contains an underscore, it is replaced by two underscores.
```

Comment:
Much nicer than naga-oil!

Reply:
tho I'm still hoping we can hide mangling from the api, leaving it as an implementation detail.
will add comment